File: Benchmarking/src/gen181.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
  }
  struct s1 {
    var p0: [s0]
    var p1: s0
  }
  @inline(never)
  func f0(_ v0: [s1], _ v1: Double) -> Double {
    let v5: s1 = v0[0]
    let v7: s0 = v5.p1
    var v10: Double = v1
    v10 = v1
    let v12: [[Double]] = v7.p0
    let v19: [Double] = v12[1]
    let v32: Double = v19[0]
    let v51: Double = v19[0]
    let v55: [Double] = [v10, v32, v1, v32, v32, v51, v10]
    var v62: [Double] = v55
    let v132: Double = v62[1]
    var v145: Double = v132
    v62[3] = v32
    return v145
  }
  func benchmark() {
    let v0: [s1] = [s1(p0: [s0(p0: [[0.0], [1.0]], p1: [[2.0], [3.0], [4.0]]), s0(p0: [[5.0], [6.0]], p1: [[7.0], [8.0], [9.0]])], p1: s0(p0: [[10.0], [11.0]], p1: [[12.0], [13.0], [14.0]])), s1(p0: [s0(p0: [[15.0], [16.0]], p1: [[17.0], [18.0], [19.0]]), s0(p0: [[20.0], [21.0]], p1: [[22.0], [23.0], [24.0]])], p1: s0(p0: [[25.0], [26.0]], p1: [[27.0], [28.0], [29.0]]))]
    let v1: Double = 30.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 30.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen181.scala
import java.lang.System.nanoTime
import scala.collection.immutable.Vector
object Gen extends App {
  case class s0 (
    p0: Vector[Vector[Double]],
    p1: Vector[Vector[Double]]
  )
  case class s1 (
    p0: Vector[s0],
    p1: s0
  )
  @noinline
  def f0(v0: Vector[s1], v1: Double): Double = {
    val v5: s1 = v0(0)
    val v7: s0 = v5.p1
    var v10: Double = v1
    v10 = v1
    val v12: Vector[Vector[Double]] = v7.p0
    val v19: Vector[Double] = v12(1)
    val v32: Double = v19(0)
    val v51: Double = v19(0)
    val v55: Vector[Double] = Vector(v10, v32, v1, v32, v32, v51, v10)
    var v62: Vector[Double] = v55
    val v132: Double = v62(1)
    var v145: Double = v132
    v62 = v62.updated(3, v32)
    v145
  }
  def benchmark(): Unit = {
    val v0: Vector[s1] = Vector(s1(Vector(s0(Vector(Vector(0.0), Vector(1.0)), Vector(Vector(2.0), Vector(3.0), Vector(4.0))), s0(Vector(Vector(5.0), Vector(6.0)), Vector(Vector(7.0), Vector(8.0), Vector(9.0)))), s0(Vector(Vector(10.0), Vector(11.0)), Vector(Vector(12.0), Vector(13.0), Vector(14.0)))), s1(Vector(s0(Vector(Vector(15.0), Vector(16.0)), Vector(Vector(17.0), Vector(18.0), Vector(19.0))), s0(Vector(Vector(20.0), Vector(21.0)), Vector(Vector(22.0), Vector(23.0), Vector(24.0)))), s0(Vector(Vector(25.0), Vector(26.0)), Vector(Vector(27.0), Vector(28.0), Vector(29.0)))))
    val v1: Double = 30.0
    val start = nanoTime()
    var result: Double = 30.0
    (1 to 1000).foreach { _ =>
      result = f0(v0, v1)
    }
    val end = nanoTime()
    println(result)
    println(end - start)
  }
  benchmark()
}